packages/jest-cli/src/reporters/BaseReporter.js

```diff
@@ -28,20 +28,20 @@ class BaseReporter {
     results: AggregatedResult,
   ) {}
 
+  onTestStart(
+    config: Config,
+    path: Path,
+    runnerContext: RunnerContext,
+  ) {}
+
   onRunComplete(
     config: Config,
     aggregatedResults: AggregatedResult,
     runnerContext: RunnerContext,
   ): ?Promise<any> {}
 
   _write(string: string) {
-    // If we write more than one character at a time it is possible that
-    // node exits in the middle of printing the result.
-    // If you are reading this and you are from the future, this might not
-    // be true any more.
-    for (let i = 0; i < string.length; i++) {
-      process.stderr.write(string.charAt(i));
-    }
+    process.stderr.write(string);
```

Comment:
i tested it on www, but not by failing that many tests.
i believe [this line](https://github.com/facebook/jest/blob/master/packages/jest-cli/src/cli/index.js#L41) should have fixed it. (`process.on('exit')`).
we should definitely test it more thoroughly though.